src/rgw/rgw_op.cc

```diff
@@ -2481,6 +2481,12 @@ void RGWDeleteBucket::execute()
   }
 
   op_ret = store->delete_bucket(s->bucket, ot);
+
+  if (op_ret == -ECANCELED && !store->is_meta_master()) {
```

Comment:
@yehudasa @cbodley was able to reproduce the multiple deletes going to master zone and master raising a -125; but no debug logs; so yes we should handle this  case too